app/models/recipe_ingredient.rb

```diff
@@ -1,3 +1,8 @@
-class RecipeIngredient < ApplicationRecord
+class RecipeIngredient < ApplicationRecord 
+  belongs_to :recipe
+  belongs_to :ingredient
 
+  def ingredient_exist?(recipe_id, ingredient_id)
+    RecipeIngredient.where(recipe_id: recipe_id, ingredient_id: ingredient)
```

Comment:
This is a great option for checking for existence. Another cool Rails trick to do this type of check is to include this validation on a joins table:
`validates :recipe_id, uniqueness: { scope: :ingredient_id }`
This will make sure that there isn't a duplicate joins record. It's best practice to also back this up with database constraints. We don't worry about this much at Turing but just so you're aware, here's a short article describing this concept and it includes an explanation about how to set up a db index that enforces uniqueness for the join record. https://boringrails.com/tips/rails-unique-scope

If you use this validation, be sure to test it! There's shouldamatchers options to test uniqueness with scope